specs/refactoring_of_node_monitoring_flows_into_node_agent.adoc

```diff
@@ -0,0 +1,223 @@
+  get the configs generated from jinja templates given the appropriate parameters.
+
+== Problem description
+
+The idea behind having node-agent and node-monitoring seperated into a seperate
+packages was to facilitate modularity in tendrl stack such that users of tendrl
+stack can have the freedom to choose whether or not to have monitoring stack
+along with tendrl core stack. But, this pulls in overhead of having to maintain
+greater number of services.So, the idea is to now consolidate the services as
+much as possible and as part of this idea, this spec proposes consolidation of
+node-monitoring into node-agent.
+
+== Use Cases
+
+Admin creates/imports a cluster using/into tendrl and has monitoring enabled
+out of the box.
+
+== Proposed change
+
+* The generic flow as in:
+
+----
+https://github.com/Tendrl/node-monitoring/blob/master/tendrl/node_monitoring/flows/configure_collectd/__init__.py
+----
+
+and a genric command to generate config file from template as in:
+
+----
+https://github.com/Tendrl/node-monitoring/blob/master/tendrl/node_monitoring/commands/config_manager.py
+----
+
+will be moved to node-agent.
+
+* The templates currently in:
+
+----
+https://github.com/Tendrl/node-monitoring/tree/master/tendrl/node_monitoring/templates
+----
+
+and the plugins currently in:
+
+----
+https://github.com/Tendrl/node-monitoring/tree/master/tendrl/node_monitoring/plugins
+----
+
+will be moved into node-agent.
+
+* Earlier the jobs for intiating the flows to configure collectd on nodes were
+  initiated by performance-monitoring node. But for now(momentarily) as a final
+  step of import, create and expand cluster, this will be intiated on all nodes
+  of cluster.
+
+  ----
+  Note:
+  Initiating configuration on cluster nodes will involve invoking configuration
+  flow after completion of import cluster flow:
+  https://github.com/Tendrl/performance-monitoring/blob/master/tendrl/performance_monitoring/utils/util.py#L41
+  ----
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+A new flow definition will be added to node-agent under its namespace as below:
+
+----
+namespace.node_agent:
+  flows:
+    ConfigureCollectd:
+      atoms:
+        - tendrl.objects.Node.atoms.Cmd
+      help: "Execute given command on given node"
+      enabled: true
+      inputs:
+        mandatory:
+          - Node.fqdn
+          - plugin_name
+          - plugin_conf_params
+          - Service.name
+      post_run:
+        - tendrl.objects.Service.atoms.CheckServiceStatus
+      pre_run:
+        - tendrl.objects.Node.atoms.CheckNodeUp
+      run: tendrl.node_agent.flows.ConfigureCollectd
+      type: Create
+      uuid: dc8fff3a-34d9-4786-9282-55eff6abb6c4
+      version: 1
+tendrl_schema_version: 0.3
+----
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+None
+
+==== Notifications/Monitoring impact:
+
+node-monitoring project will be deprecated.
+
+==== Tendrl/common impact:
+None
+
+==== Tendrl/node_agent impact:
```

Comment:
Provide details of triggering "ConfigureCollectd" after success if ImportCluster